### PR TITLE
Skin updates

### PR DESCRIPTION
**Pull request by poelzi**
I was putting some love into the deer skin and was able to skin the menubar after some small changes. Since we are skinning, having a default menubar feels like a foreign object, so I consider this 2.3 worthy.

* Deere: option for a larger UV master.
* Indicate green/yellow area in the UV meters
* ~~Allow skinning of menubar~~ #3372 
* ~~Skin Deer Menubar~~ future PR by @ronso0 
* Small adjustments of the alpha value for track colors for better visibility (all skins)

Screenshot_20201111_002045

### CODE REVIEW

**Comment by daschuer** [comment]
I cannot see a difference in case of Gnome or Unity. 
Can you add also a "Before" Screenshot?

**Comment by daschuer** [comment]
I am in doubt that the large VU Meter should be a preference option. The designer should decide.

**Comment by daschuer** [comment]
OK there is a difference with Deere:
grafik

**Comment by poelzi** [comment]
> I am in doubt that the large VU Meter should be a preference option. The designer should decide.

I differ. Some people might like to have same sized UV meters and this is a simple option that does not much harm.

**Comment by poelzi** [comment]
> I cannot see a difference in case of Gnome or Unity.
> Can you add also a "Before" Screenshot?

It depends on your system colors. If you run a dark skin it might look ok, on a light skin, at least on KDE, it looks ugly.

**Comment by Holzhaus** [comment]
> > I cannot see a difference in case of Gnome or Unity.
> > Can you add also a "Before" Screenshot?
> 
> It depends on your system colors. If you run a dark skin it might look ok, on a light skin, at least on KDE, it looks ugly.

https://mixxx.org/screenshots/

File: res/skins/Deere/style.qss [hunk 1]
```diff
@@ -2190,3 +2201,28 @@ WEffectSelector {
...
+
+#MainMenu QMenu::indicator:checked,
+#MainMenu QMenu::indicator:checked:selected {
+  image: url(skin:/icon/ic_library_checkmark.svg);
+}
+
+#MainMenu QMenu::indicator:unchecked {
+  background: none;
+  border: 1px solid rgba(151,151,151,128);
+}
```
Comment (ronso0) [inline res/skins/Deere/style.qss:2228]: Can be set together with `#LibraryContainer QTableView::indicator:unchecked` in line 276

File: res/skins/Deere/style.qss [hunk 2]
```diff
@@ -2190,3 +2201,28 @@ WEffectSelector {
...
+
+#MainMenu QMenu::item:unchecked {
+  padding: 5px 13px 5px 10px;
+}
+
+#MainMenu QMenu::indicator:checked,
+#MainMenu QMenu::indicator:checked:selected {
+  image: url(skin:/icon/ic_library_checkmark.svg);
+}
```
Comment (ronso0) [inline res/skins/Deere/style.qss:2223]: Can be moved up to ```
#LibraryContainer QTableView::indicator:checked,
#LibraryContainer QTableView::indicator:checked:selected
```, only the border colors differ.

**Comment by ronso0** [comment]
> OK there is a difference with Deere:
> grafik

One reason I didn't start working on styles for DlgtrackInfo is that once you start overwriting native styles you need to take care of  _all_ other styles, too, and make sure they look good on all platforms. Thus my library menu style PR has grown enormously and has taken ages to finish.

**Comment by ronso0** [comment]
Please rebase on 2.3 so crash fixes from #3144 are included and this PR is testable.

**Comment by ronso0** [comment]
> > I am in doubt that the large VU Meter should be a preference option. The designer should decide.
> 
> I differ. Some people might like to have same sized UV meters and this is a simple option that does not much harm.

@Be-ing ??

**Comment by poelzi** [comment]
@ronso0 I felt your pain yesterday. So, maybe as a quick workaround we could just set the colors of the menus and a border.
With changing check symbols the pain started which might have to do with using SVG as icons, or at least, does not make it easier.

**Comment by poelzi** [comment]
This is how 2.3 Deer looks with KDE on a light skin.
Screenshot_20201111_143147

**Comment by ronso0** [comment]
> @ronso0 I felt your pain yesterday. So, maybe as a quick workaround we could just set the colors of the menus and a border.
> With changing check symbols the pain started which might have to do with using SVG as icons, or at least, does not make it easier.

maybe take a look at a system theme and check how menus and checkboxes are styled there.
I doubt it fails because of SVGs, those work in the track menu. You may need to set the indicator size explicitely in css.

**Comment by poelzi** [comment]
@ronso0 do you have some time helping on this one ? I have so many open points that this styling thing drives me nuts. You wrote that you did some styling of the menus already ?

**Comment by ronso0** [comment]
I think it'll take an entire evening to look into this and supply comprehensive styles, let alone applying this to all skins.
Unfortunately I don't have time for this until sunday.

**Comment by poelzi** [comment]
@ronso0 even if it takes you 2 weeks, I would be super happy getting your skilled styling on board here. I tried different things this evening but could not get something look nice everywhere :(

**Comment by ronso0** [comment]
Are you asking just about the styling or about purging quirks like the one @daschuer experienced?

**Comment by ronso0** [comment]
@poelzi 
I suggest you split this up into smaller chunks to get the finished parts merged
* Deere design
* track color opacity
* menubar styling, c++ requirements (no skin changes): we need to check what happens on macOS, Windows and various linux DEs, and since we're short on active Win/macOS testers it would be better to have style-capable builds available and users can test stylesheets without having to build Mixxx on their own
* WIP menubar stylesheets

**Comment by Be-ing** [comment]
I agree with @daschuer and @Holzhaus. I am not a fan of the big meter option in Deere. It looks out of place to me and I don't think the option is really helpful. Having more options is not inherently good.

**Comment by ronso0** [comment]
> I agree with @daschuer and @Holzhaus. I am not a fan of the big meter option in Deere. It looks out of place to me and I don't think the option is really helpful.

I think it could be a useful addition and we should make it the default.
But it's to big IMO compared to the channel meters. If we limit it to half the height of the mixer it looks good.
Also let's include the colored meter backpaths.

**Comment by ronso0** [comment]
> * menubar styling, c++ requirements (no skin changes): we need to check what happens on macOS, Windows and various linux DEs, and since we're short on active Win/macOS testers it would be better to have style-capable builds available and users can test stylesheets without having to build Mixxx on their own

IMO we should definitively pick that for 2.3 and subsequently style the menubars in point releases.
This feature has been requested for ages.

**Comment by Be-ing** [comment]
> IMO we should definitively pick that for 2.3 and subsequently style the menubars in point releases.

I think that effort would be better spent getting rid of the menu bar.

**Comment by ronso0** [comment]
> > IMO we should definitively pick that for 2.3 and subsequently style the menubars in point releases.
> 
> I think that effort would be better spent getting rid of the menu bar.

There's already the next step in progress #3189 getting rid of the menu bar. After that we'll have to rework The Menu (however that will look like) and make it accessible for screen readers.
That'll take ages, and 2.3 will be the stable version for ~2 years :: looking at our release tempo o_o`
That said, it doesn't make much sense to me accepting the current situation while the styling groundwork is already done here.

@poelzi I you don't have time to split off the menubar theming prerequisities I'll open a separate PR with that commit and finish it in the next weeks.

**Comment by ronso0** [comment]
I picked up 851adbe94ba09be622346058d071d54fee102861 for #3372

**Comment by ronso0** [comment]
Alright, I managed to adjust sizes and margins of the menu item, checkboxes & icons by using `em` 
image
PR coming soon...

At first I was puzzled why the font-sizes wouldn't react to OS font size changes, then I realized it would only obey font settings made in `qtconfig` / `Qt5 Settings` (thus the italic font above).
I think we should not mess with the menu font settings since those are important accessibility parameters users need to be able to adjust by themselves (and we don't have to worry about the layout as we have to in the Mixxx skins), so either omitting menu font sizes in the stylesheets or setting them to `1em` ensures readable menus and should not create accessibility regressions.

By setting some aburd font I also noticed that Mixxx font settings are not complete: Qt config could screw up GUI buttons, labels, table headers etc. I think there are only a few adjustments missing, so I guess I can fix all of them for 2.3, too.

**Comment by poelzi** [comment]
@ronso0 I removed the skin styling commits here and only left the large VU meter option in deer, the dimmed background image in VU and the increase in background color in the libraries. I adjusted them individually depending on how clear the background is visible.

**Comment by ronso0** [comment]
re large VU meter in Deere:
some tweaking is necessary to align it with some channel controls. https://github.com/ronso0/mixxx/commit/c9dc5002f2cbb1cfd5455b0721b047dd48a4f951 shoud do the trick.

re track colors:
LGTM, could you also increase it in Shade?

**Comment by ronso0** [comment]
btw I made some good progress with the menubar styling (Deere only as of now).
I hope to finish that and apply the styles to the other skins this week.

**Comment by ronso0** [comment]
> re large VU meter in Deere:
> some tweaking is necessary to align it with some channel controls. [ronso0@c9dc500](https://github.com/ronso0/mixxx/commit/c9dc5002f2cbb1cfd5455b0721b047dd48a4f951) shoud do the trick.
> 
> re track colors:
> LGTM, could you also increase it in Shade?

btw to keep the commit history clean please make use of `git --fixup ...` and `git rebase -i --autosquash` for this PR because a) it is rather small and b) more about the effect than about reviewable code.
thx!

**Comment by poelzi** [comment]
@ronso0 I tried your patch but the spacing on top just did not make sense for me. The purpose of the large VU option is, to use as much space as possible for the main VU meter. Only the spacer on the bottom for crossfader assignment was required otherwise it looked strange.

**Comment by ronso0** [comment]
@poelzi I'll take another look at this during the weekend.

**Comment by ronso0** [comment]
@poelzi I know the Deere mixer can be a pita to work with but in this branch the alignment of the large Main VU is horrible in some configurations. We should not sacrifice asthetics just for having a large VU meter.
Please try to fix that somehow with conditional spacers.
image
image

**Comment by poelzi** [comment]
I was not able to reproduce this on my branch, it always looks ok

**Comment by ronso0** [comment]
this is the config I refer to: 4 decks, separate waveforms
image
how does that look like for you?

please just add some spacers for this config and we can merge.

**Comment by ronso0** [comment]
with https://github.com/ronso0/mixxx/commit/95db5d72178c3d2b02a739eb8b6e71db04ad6de4 it looks like this, there's an offset only when there's no quick effect configured
image

pick this, double-check that it looks okay to you and then we can merge.

File: res/skins/Deere/mixer_column_master_vu_4decks.xml [hunk 3]
```diff
@@ -8,86 +8,60 @@
...
-              <BindProperty>visible</BindProperty>
-            </Connection>
+            <Layout>vertical</Layout>
+            <SizePolicy>min,me</SizePolicy>
+            <MaximumSize>60,300</MaximumSize>
+            <Children>
+              <SingletonContainer>
+                <ObjectName>StereoVUMeterMaster</ObjectName>
+              </SingletonContainer>
+            </Children>
```
Comment (ronso0) [inline res/skins/Deere/mixer_column_master_vu_4decks.xml:32]: https://github.com/ronso0/mixxx/commit/95db5d72178c3d2b02a739eb8b6e71db04ad6de4

**Review by ronso0 — CHANGES_REQUESTED** [review]

**Comment by poelzi** [comment]
@ronso0 done, looks good here

**Comment by ronso0** [comment]
Okido, the rest LGTM
Thanks!

**Pull request merged**